fhirzeug/generators/python_pydantic/templates/resource_header.py

```diff
@@ -24,26 +27,187 @@ def check_at_least_one(cls, values):
+    """Build a root validator that validates a primitive field.
+
+    Root validator is used in order to have access to all other (already validated)
+    fields. `skip_on_failure` is set in order to avoid validating fields that
+    might not be cleaned.
+    """
+
+    @pydantic.root_validator(pre=True, skip_on_failure=True, allow_reuse=True)
+    def _validator(
+        cls, values: typing.Dict[str, typing.Any]
+    ) -> typing.Dict[str, typing.Any]:
+        # Check if both field and extension are set.
+        # If field or extension is not set, we do not need to validate the consistency
+        # between them.
+        # Note: that might not be the case anymore when we will also validate cardinality.
+
+        # Field can either be present as the real field name or its alias
+        inner_field_name = field_name
+        if inner_field_name not in values:
+            inner_field_name = alias_generator(inner_field_name)
+            if inner_field_name not in values:
+                return values
+
+        # Extension can either be present as the real extension name or its alias
+        extension_name = field_name + _EXTENSION_SUFFIX
+        if extension_name not in values:
+            extension_name = alias_generator(extension_name)
+            if extension_name not in values:
+                return values
+
+        # Get field and extension values
+        field_value = values[inner_field_name]
+        extension_value = values[extension_name]
+
+        # Validate them and get validated values
+        validated_field_value, validated_extension_value = _validate_primitive_field(
+            field_value, extension_value
+        )
+
+        # Assign new values and return
+        values[inner_field_name] = validated_field_value
+        values[extension_name] = validated_extension_value
+        return values
+
+    return _validator
+
+
+def _validate_primitive_field(
+    initial_field_value: typing.Any, extension_field_value: typing.Any
+) -> typing.Tuple[typing.Any, typing.Any]:
+    """Validate the consistency of a primitive field or list-of-primitives field.
+
+    Note: `initial_field_value` refer to the primitive field that is extended
+          and `extension_field_value` refer to the extension field.
+
+    Validators :
+        - from https://www.hl7.org/fhir/json.html#primitive :
+          "In the case where the primitive element may repeat, it is represented
+          in two arrays. JSON null values are used to fill out both arrays so
+          that the id and/or extension are aligned with the matching value in the
+          first array."
+          "Note: when one of the repeating elements has no value, it is represented
+          in the first array using a null. When an element has a value but no
+          extension/id, the second array will have a null at the position of that
+          element."
+
+        - TODO: validate cardinality ? Make initial field optional ?
```

Comment:
I have completed the `TODO`. We have a case to validate when a field is required (cardinality 1 or 1..*). From my understanding of FHIR specs, it is possible to provide only an extension instead of the primitive value.
But currently this is not done. I just created an issue about it : https://github.com/skalarsystems/fhirzeug/issues/70